vm-wasmi/src/vm.rs

```diff
@@ -505,7 +505,22 @@ where
             .and_then(Extern::into_func)
             .ok_or(WasmiVMError::FunctionNotFound)?;
 
-        export.call(self.0.as_context_mut(), &params, &mut result)?;
+        // Avoid DOS caused by infinite recursion.  See
+        // https://github.com/JumpCrypto/security-research/blob/master/advisories/2023-003-cosmwasm.md?ref=jumpcrypto.com
+        let mut ctx = self.0.as_context_mut();
+        let depth = ctx.data_mut().call_depth_mut();
+        if *depth >= 2 {
```

Comment:
maybe we can make this configurable? I'm just wondering if there could be a reason for this to recurse on a good case?

Reply:
If you mean configured at run-time than in principle yes but I just don’t know where that option should be specified.

Reply:
i am ok with constant. error case is here. clear when  to change. cosmos people also have constant.